proposals/NNNN-Deprecated-interfaces-markup.md

```diff
@@ -0,0 +1,132 @@
+#else
+#pragma message("WARNING: You need to implement DEPRECATED for this compiler")
+#define DEPRECATED
+#endif
+#endif
+```
+
+It will generate compile time warning in with c++98 or c++11 or c++14.
+
+### Usage : 
+
+#### C++ Code :
+```cpp
+DEPRECATED void bar() {}
+int main() {
+  bar();
+}
+```
+
+#### Compilers output : 
+
+##### G++ :
+```bash
+$ g++ --std=c++11 -Werror -Wno-error=deprecated-declarations ./main.cc                            
+./main.cc: In function ‘int main()’:
+./main.cc:85:5: warning: ‘void bar()’ is deprecated (declared at ./main.cc:19) [-Wdeprecated-declarations]
+     bar();
+     ^
+./main.cc:85:9: warning: ‘void bar()’ is deprecated (declared at ./main.cc:19) [-Wdeprecated-declarations]
+     bar();
+
+$  g++ --std=c++98 -Werror -Wno-error=deprecated-declarations ./main.cc 
+./main.cc: In function ‘int main()’:
+./main.cc:85:5: warning: ‘void bar()’ is deprecated (declared at ./main.cc:19) [-Wdeprecated-declarations]
+     bar();
+     ^
+./main.cc:85:9: warning: ‘void bar()’ is deprecated (declared at ./main.cc:19) [-Wdeprecated-declarations]
+     bar();
+         ^
+```
+
+##### Clang : 
+```bash
+$ clang --std=c++98 -Werror -Wno-error=deprecated-declarations ./main.cc  
+./main.cc:23:5: warning: 'bar' is deprecated [-Wdeprecated-declarations]
+    bar();
+    ^
+./main.cc:19:17: note: 'bar' has been explicitly marked deprecated here
+DEPRECATED void bar() {}
+                ^
+1 warning generated.
+
+$ clang --std=c++11 -Werror -Wno-error=deprecated-declarations ./main.cc  
+./main.cc:23:5: warning: 'bar' is deprecated [-Wdeprecated-declarations]
+    bar();
+    ^
+./main.cc:19:17: note: 'bar' has been explicitly marked deprecated here
+DEPRECATED void bar() {}
+                ^
+1 warning generated.
+```
+
+Full usage example is available https://github.com/alexkutsan/RnD_fun/blob/master/deprecated/main.cc 
+
+Also doxygen `@deprecated` command should be used for documentation:
+```cpp
+/**
+ * @brief The Y class containt one actual and one deprecated method
+ */
+class Y {
+
+public:
+
+    /**
+     * @brief foo not deprecated function of not deprecated class
+     */
+    void foo(){}
+
+    /**
+     * @brief bar deprecated function of not deprecated class
+     * @deprecated
+     */
+    DEPRECATED int bar(){}
+};
+```
+
+![Doxygen result](/assets/proposals/nnnn-deprecated-interfaces-markup/deprecated_doxygen_example.png)                                                                                                                                                                                                                                                    
+
+## Impact on existing code
+- No changes in behaviour should be introduced. 
+- Some existing SDL core methods should be marked as deprecated. If methods used in code it will cause compile time warnings.
+
+## Alternatives
+As alternative it is possible to remove deprecated methods, but it will break backward compatibility. 
```

Comment:
Can you please remove the period after "compatibility"?  I believe this entire section should be one sentence.

The sentence should end "...but it will break backward compatibility of 3rd party developers' code with newer versions of SDL."